src/darwin/Framework/CHIP/MTRDeviceController_Internal.h

```diff
@@ -256,6 +256,8 @@ NS_ASSUME_NONNULL_BEGIN
 - (MTRDevice *)deviceForNodeID:(NSNumber *)nodeID;
 - (void)removeDevice:(MTRDevice *)device;
 
+- (NSNumber * _Nullable)syncGetCompressedFabricID;
```

Comment:
Followup: Just cache this when controller starts, and nix the sync dispatch and the workarounds we had to make for it in this code.